sparse_ho/tests/test_criterion.py

```diff
@@ -1,81 +1,92 @@
 X, y, _ = make_correlated_data(
-    n_samples, n_features, corr=corr, snr=snr, random_state=42)
-
-alpha_max = (np.abs(X.T @ y)).max() / n_samples
-tol = 1e-8
-
-estimator = sklearn.linear_model.Lasso(
-    fit_intercept=False, max_iter=10000, warm_start=True)
-model = Lasso(estimator=estimator)
-
-log_alphas = np.log(np.geomspace(alpha_max, alpha_max / 100))
-
-
-def test_cross_val_criterion():
-    # TODO we also need to add a test for sparse matrices
-    alpha_min = alpha_max / 10
-    log_alpha_max = np.log(alpha_max)
-    log_alpha_min = np.log(alpha_min)
-    max_iter = 10000
-    n_alphas = 10
-    kf = KFold(n_splits=5, shuffle=True, random_state=56)
-
-    monitor_grid = Monitor()
-    mse = HeldOutMSE(None, None)
-    criterion = CrossVal(mse, cv=kf)
+    n_samples, n_features, corr=0.1, snr=3, random_state=42)
+sigma_star = 0.1
+
+y = np.sign(y)
+idx_train = np.arange(0, n_samples//2)
+idx_val = np.arange(n_samples//2, n_features)
+
+# Set alpha for the Lasso
+alpha_max = (np.abs(X[idx_train, :].T @ y[idx_train])).max() / n_samples
+p_alpha = 0.8
+alpha = p_alpha * alpha_max
+log_alpha = np.log(alpha)
+log_alpha_max = np.log(alpha_max)
+tol = 1e-15
+
+# Set alpha1 alpha2 for the enet
+alpha_1 = p_alpha * alpha_max
+alpha_2 = 0.1
+log_alpha1 = np.log(alpha_1)
+log_alpha2 = np.log(alpha_2)
+
+dict_log_alpha = {}
+dict_log_alpha["lasso"] = log_alpha
+dict_log_alpha["enet"] = np.array([log_alpha1, log_alpha2])
+tab = np.linspace(1, 1000, n_features)
+dict_log_alpha["wLasso"] = log_alpha + np.log(tab / tab.max())
+dict_log_alpha["logreg"] = (log_alpha - np.log(2))
+
+# Set models to be tested
+models = {}
+models["lasso"] = Lasso(estimator=None)
+models["enet"] = ElasticNet(estimator=None)
+models["wLasso"] = WeightedLasso(estimator=None)
+models["logreg"] = SparseLogreg(estimator=None)
+
+custom_models = {}
+custom_models["lasso"] = Lasso(estimator=celer.Lasso(
+    warm_start=True, fit_intercept=False))
+custom_models["enet"] = ElasticNet(
+    estimator=linear_model.ElasticNet(warm_start=True, fit_intercept=False))
+custom_models["logreg"] = SparseLogreg(
+    estimator=celer.LogisticRegression(warm_start=True, fit_intercept=False))
+
+
+# log alpha to be tested by checkgrad
+dict_list_log_alphas = {}
+dict_list_log_alphas["lasso"] = np.log(
+    np.geomspace(alpha_max/2, alpha_max/5, num=5))
+dict_list_log_alphas["wLasso"] = [
+    log_alpha * np.ones(n_features) for log_alpha in
+    dict_list_log_alphas["lasso"]]
+dict_list_log_alphas["logreg"] = np.log(
+    np.geomspace(alpha_max/5, alpha_max/40, num=5))
+dict_list_log_alphas["enet"] = [np.array(i) for i in itertools.product(
+    dict_list_log_alphas["lasso"], dict_list_log_alphas["lasso"])]
+
+list_model_crit = [
+    ('lasso',  HeldOutMSE(idx_train, idx_val)),
+    ('enet', HeldOutMSE(idx_train, idx_val)),
+    ('wLasso', HeldOutMSE(idx_train, idx_val)),
+    ('lasso', FiniteDiffMonteCarloSure(sigma_star)),
+    ('logreg', HeldOutLogistic(idx_train, idx_val))]
```

Comment:
we have a lot of code duplication across the tests.

typically things you put in the header of your tests could be factorized in a file sparse_ho/tests/common.py

Reply:
done

Reply:
done